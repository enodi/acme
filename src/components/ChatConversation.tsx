import React from "react";

import { userDataFormat } from "../models";

interface ChatConversationProps {
  conversation: userDataFormat;
}

const ChatConversation: React.FC<ChatConversationProps> = ({
  conversation,
}) => (
  <div className="chat-screen__area">
    {conversation.messages.map((message: string, index: number) => (
      <span key={`message-${index}`} className="chat-screen__bubble">
        {message}
      </span>
    ))}
  </div>
);

export default ChatConversation;
