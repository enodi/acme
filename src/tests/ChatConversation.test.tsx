import React from "react";
import { shallow } from "enzyme";

import ChatConversation from "../components/ChatConversation";
import { selectedItem } from "./fixtures";

let wrapper: any;
beforeEach(() => {
  wrapper = shallow(<ChatConversation conversation={selectedItem} />);
});

test("should render ChatConversation component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should find text rendered on chat", () => {
  expect(wrapper.find("span").text()).toEqual(selectedItem.messages[0]);
});
