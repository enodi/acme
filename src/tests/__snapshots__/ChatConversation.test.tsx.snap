// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ChatConversation component correctly 1`] = `
<div
  className="chat-screen__area"
>
  <span
    className="chat-screen__bubble"
    key="message-0"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit
  </span>
</div>
`;
